### ISSUE TITLE
False negative when accessing annotated class attributes

### ISSUE THREAD

**Issue by denballakh** [python/mypy#13103]
Consider this example:
```py
class X:
    __slots__ = ('a',)
    a: int

print(X.a) # <member 'a' of 'X' objects>
reveal_type(X.a) # builtins.int
reveal_type(X.a.__get__) # Any
    # "int" has no attribute "__get__"; maybe "__gt__", "__ge__", or "__ne__"?
```
I expect to get `member_descriptor` and some `Callable`, but instead i see two wrong reveals and one error.

I can use this bug to write unsafe code:
```py
class X:
    __slots__ = ('a',)
    a: int

x = X()
x.a = X.a # no error!
assert isinstance(x.a, int), x.a
# AssertionError: <member 'a' of 'X' objects>
```

**My Environment**
- `mypy 0.961 (compiled: no)`
- no command-line flags, no config file
- `CPython 3.10.4`
- Windows 10

**Comment by JukkaL** [comment]
Some hints:
* The slots of a class are available in the `slots` attribute of `TypeInfo`
* The relevant code is in `mypy.checkmember`. Probably look at `analyze_type_callable_member_access` and `analyze_type_type_member_access`.
* We can probably just generate an error if trying to access `Cls.x` if `x` has a slot. It's rarely the right thing to do.

**Comment by denballakh** [comment]
This might be related:
```py
class X:
    __slots__ = ('x',)
    x: int = 1

# Traceback (most recent call last):
#   File "...", line ..., in <module>
#     class X:
# ValueError: 'x' in __slots__ conflicts with class variable

reveal_type(X.x) # int
reveal_type(X().x) # int
```

**Comment by DanielGoman** [comment]
Hello :)
I'm new to open source contribution. 
If this issue is still relevant, I'd like to start working on it.
Could you please assign it to me?

**Comment by hmccarty** [comment]
My first go at contributing, let me know if you had something else in mind @JukkaL

**Comment by hmccarty** [comment]
Any updates on this?
